### PR TITLE
feat: Sync upstream with Azuredisk CSI Driver V2 as of 07/16/2021

### PR DESCRIPTION
**Pull request by nearora-msft**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: .github/workflows/linux-v2.yml [hunk 1]
```diff
@@ -24,6 +24,7 @@ jobs:
 
     - name: Run Linux Unit Tests
       run: sudo go test -covermode=count -coverprofile=profile.cov -tags azurediskv2 ./pkg/azuredisk --temp-use-driver-v2
+      sudo go test -covermode=count -coverprofile=profile.cov -tags azurediskv2 ./pkg/azdiskschedulerextender
```
Comment (edreed) [inline .github/workflows/linux-v2.yml:27]: Using flow scalars works, but might be clearer and more consistent to use literal block scalar here, e.g.
```
      run: |
        sudo go test -covermode=count -coverprofile=profile.cov -tags azurediskv2 ./pkg/azuredisk --temp-use-driver-v2
        sudo go test -covermode=count -coverprofile=profile.cov -tags azurediskv2 ./pkg/azdiskschedulerextender```

Comment (nearora-msft) [reply .github/workflows/linux-v2.yml]: Fixed

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Pull request merged**